pkg/connectors/gitlab/gitlabConnector.go

```diff
@@ -0,0 +1,230 @@
+	Error   error
+}
+
+func New(deps *GitlabConnectorDependencies) *GitlabConnector {
+	return &GitlabConnector{
+		client: deps.Client,
+	}
+}
+
+func (gc *GitlabConnector) Get(args []string) (int, error) {
+	projectsChan := gc.getAllProjects(args)
+	reposFetchCounter := 0
+	gitlabJsonObject := make(map[string]*GitlabGroup)
+
+	for project := range projectsChan {
+		if project.Error != nil {
+			return reposFetchCounter, project.Error
+		}
+
+		reposFetchCounter += 1
+		err := gc.addProject(gitlabJsonObject, project.Project)
+		if err != nil {
+			fmt.Println(err)
+			continue
+		}
+
+		err = gc.processPipelineFiles(gitlabJsonObject, project.Project)
+		if err != nil {
+			fmt.Println(err)
+			continue
+		}
+	}
+
+	gitlabJson, err := json.MarshalIndent(gitlabJsonObject, "", "  ")
+	if err != nil {
+		return reposFetchCounter, err
+	}
+
+	alleroHomedir := fileManager.GetAlleroHomedir()
+	return reposFetchCounter, fileManager.WriteToFile(fmt.Sprintf("%s/repo_files/gitlab.json", alleroHomedir), gitlabJson)
+}
+
+func (gc *GitlabConnector) processPipelineFiles(gitlabJsonObject map[string]*GitlabGroup, project *gitlab.Project) error {
+	workflowFilesChan, _ := gc.getPipelineFiles(project)
+	var processingError error
+
+	for workflowFile := range workflowFilesChan {
+		byteContent, _, err := gc.client.RepositoryFiles.GetRawFile(project.ID, workflowFile.Filename, &gitlab.GetRawFileOptions{})
+		if err != nil {
+			processingError = fmt.Errorf("failed to get content for file %s from repository %s", workflowFile.Filename, project.PathWithNamespace)
+			continue
+		}
+
+		jsonContentBytes, err := connectors.YamlToJson(byteContent)
+		if err != nil {
+			processingError = err
+			continue
+		}
+
+		jsonContent := make(map[string]interface{})
+		err = json.Unmarshal(jsonContentBytes, &jsonContent)
+		if err != nil {
+			processingError = err
+			continue
+		}
+
+		workflowFile.Content = jsonContent
+		// filenameWithoutPostfix := strings.Split(workflowFile.Filename, ".")[0]
+
+		var filenameWithoutPostfix string
+		parts := strings.Split(workflowFile.Filename, ".")
+		if len(parts) > 0 {
+			parts = parts[:len(parts)-1]
+			filenameWithoutPostfix = strings.Join(parts, ".")
+		}
+
+		if workflowFile.Origin == "gitlab_ci" {
+			gitlabJsonObject[project.Namespace.Name].Projects[project.Path].GitlabCi[filenameWithoutPostfix] = workflowFile
+		} else if workflowFile.Origin == "jfrog_pipelines" {
+			gitlabJsonObject[project.Namespace.Name].Projects[project.Path].JfrogPipelines[filenameWithoutPostfix] = workflowFile
+		} else {
+			processingError = fmt.Errorf("unsupported CICD platform %s for file %s from repository %s", workflowFile.Origin, workflowFile.Filename, project.PathWithNamespace)
+			continue
+		}
+	}
+
+	return processingError
+}
+
+func (gc *GitlabConnector) getPipelineFiles(project *gitlab.Project) (chan *PipelineFile, error) {
+	pipelineFilesChan := make(chan *PipelineFile)
+
+	var getEntitiesErr error
+	go func() {
+		defer func() {
```

Comment:
please check if possible:
`defer close(pipelineFilesChan)`
There are more places with this comment